src/System.IO.FileSystem/tests/FileStream/Dispose.cs

```diff
@@ -58,6 +64,94 @@ protected override void Dispose(bool disposing)
             }
         }
 
+
+        [Fact]
+        public void Dispose_ThrowDuringFlushWriteBuffer_TrueArgThrows()
+        {
+            RemoteInvoke(() =>
+            {
+                string fileName = GetTestFilePath();
+                using (FileStream fscreate = new FileStream(fileName, FileMode.Create))
+                {
+                    fscreate.WriteByte(0);
+                }
+                bool writeDisposeInvoked = false;
+                Action<bool> writeDisposeMethod = (disposing) =>
+                {
+                    writeDisposeInvoked = true;
+                };
+                using (var fsread = new FileStream(fileName, FileMode.Open, FileAccess.Read))
+                {
+                    Action act = () => // separate method to avoid JIT lifetime-extension issues
```

Comment:
"JIT lifetime-extension issues" .. I see copied from existing tests  .. @stephentoub  do you know what this is for? Is it because in CHK builds, even if I set reference to null, and run GC.Collect/GC.WFPF, it's not necessarily going to trigger the finalizer?